go-controller/pkg/node/gateway_localnet.go

```diff
@@ -74,27 +74,43 @@ func (n *OvnNode) initLocalnetGateway(subnet *net.IPNet, nodeAnnotator kube.Anno
 	if err != nil {
 		return err
 	}
-
-	var gatewayIP, gatewayNextHop net.IP
-	var gatewaySubnetMask net.IPMask
-	if utilnet.IsIPv6CIDR(subnet) {
-		gatewayIP = net.ParseIP(v6localnetGatewayIP)
-		gatewayNextHop = net.ParseIP(v6localnetGatewayNextHop)
-		gatewaySubnetMask = net.CIDRMask(v6localnetGatewaySubnetPrefix, 128)
-	} else {
-		gatewayIP = net.ParseIP(v4localnetGatewayIP)
-		gatewayNextHop = net.ParseIP(v4localnetGatewayNextHop)
-		gatewaySubnetMask = net.CIDRMask(v4localnetGatewaySubnetPrefix, 32)
+	// Flush any addresses on localnetGatewayNextHopPort first
+	if err := util.LinkAddrFlush(link); err != nil {
+		return err
 	}
-	gatewayIPCIDR := &net.IPNet{IP: gatewayIP, Mask: gatewaySubnetMask}
-	gatewayNextHopCIDR := &net.IPNet{IP: gatewayNextHop, Mask: gatewaySubnetMask}
 
-	// Flush any addresses on localnetBridgeNextHopPort and add the new IP address.
-	if err = util.LinkAddrFlush(link); err == nil {
-		err = util.LinkAddrAdd(link, gatewayNextHopCIDR)
-	}
-	if err != nil {
-		return err
+	var gatewayIfAddrs []*net.IPNet
+	var nextHops []net.IP
+	for _, hostSubnet := range hostSubnets {
+		var gatewayIP, gatewayNextHop net.IP
+		var gatewaySubnetMask net.IPMask
+		if utilnet.IsIPv6CIDR(hostSubnet) {
+			gatewayIP = net.ParseIP(v6localnetGatewayIP)
+			gatewayNextHop = net.ParseIP(v6localnetGatewayNextHop)
+			gatewaySubnetMask = net.CIDRMask(v6localnetGatewaySubnetPrefix, 128)
+		} else {
+			gatewayIP = net.ParseIP(v4localnetGatewayIP)
+			gatewayNextHop = net.ParseIP(v4localnetGatewayNextHop)
+			gatewaySubnetMask = net.CIDRMask(v4localnetGatewaySubnetPrefix, 32)
+		}
+		gatewayIfAddr := &net.IPNet{IP: gatewayIP, Mask: gatewaySubnetMask}
+		gatewayNextHopIfAddr := &net.IPNet{IP: gatewayNextHop, Mask: gatewaySubnetMask}
+		gatewayIfAddrs = append(gatewayIfAddrs, gatewayIfAddr)
+		nextHops = append(nextHops, gatewayNextHop)
+
+		if err = util.LinkAddrAdd(link, gatewayNextHopIfAddr); err != nil {
+			return err
+		}
+		if utilnet.IsIPv6CIDR(hostSubnet) {
+			// TODO - IPv6 hack ... for some reason neighbor discovery isn't working here, so hard code a
+			// MAC binding for the gateway IP address for now - need to debug this further
+			err = util.LinkNeighAdd(link, gatewayIP, macAddress)
```

Comment:
During the discussion about LinkNeighAdd vs Set, we had decided to make a call to LinkNeighExists before making the call to LinkNeighAdd. That would prevent the log message in case the entry already exists in case of restarts etc. Sorry missed this earlier.